app/src/main/java/com/flowfoundation/wallet/manager/walletconnect/WalletConnectDelegate.kt

```diff
@@ -67,53 +112,93 @@ internal class WalletConnectDelegate : SignClient.WalletDelegate {
     ) {
         logd(TAG, "onSessionProposal() sessionProposal json:${Gson().toJson(sessionProposal)}")
         logd(TAG, "onSessionProposal() verifyContext json:${Gson().toJson(verifyContext)}")
-        val activity = BaseActivity.getCurrentActivity() ?: return
+        
         processedRequestIds.clear()
-        try {
-            uiScope {
-                with(sessionProposal) {
-                    val approve = if (WalletManager.isEVMAccountSelected()) {
-                        if (isShowMoveDialog()) {
-                            MoveDialog().showMove(activity.supportFragmentManager, description)
-                        }
-                        EvmRequestAccountDialog().show(
-                            activity.supportFragmentManager,
-                            EVMDialogModel(
+        isSessionSettled = false
+        isAuthnComplete = false
+        
+        // Set the redirect URL synchronously before showing the dialog
+        pendingRedirectUrl = if (sessionProposal.redirect.isNotEmpty()) {
+            logd(TAG, "Using provided redirect URL: ${sessionProposal.redirect}")
+            sessionProposal.redirect
+        } else {
+            logd(TAG, "No redirect provided, using DApp URL: ${sessionProposal.url}")
+            sessionProposal.url
```

Comment:
We need double check on this, for example:
if user comes from  flow port A, now if we navigate user back with flow port url.
Will it open flow port B ? But our app was just authn the flow port A.